src/export/StructureDefinitionExporter.ts

```diff
@@ -420,6 +411,124 @@ export class StructureDefinitionExporter implements Fishable {
     });
   }
 
+  private setContext(structDef: StructureDefinition, fshDefinition: Extension) {
+    // Set context to everything by default, but users can override w/ Context keyword or rules, e.g.
+    // Context: Observation
+    // ^context[0].type = #element
+    // ^context[0].expression = "Patient"
+    if (fshDefinition.contexts.length > 0) {
+      structDef.context = [];
+      fshDefinition.contexts.forEach(extContext => {
+        if (extContext.isQuoted) {
+          structDef.context.push({
+            expression: extContext.value,
+            type: 'fhirpath'
+          });
+        } else {
+          // this could be the path to an element on some StructureDefinition
+          // it could be specified either as a url, a path, or a url with a path
+          // url: http://example.org/Some/Url
+          // path: resource-name-or-id.some.path
+          // url with path: http://example.org/Some/Url#some.path
+          // we split on # to make contextItem and contextPath, so if contextPath is not empty, we have a url with a path
+          // otherwise, check if contextItem is a url or not. if it is not a url, we have a fsh path that starts with a name or id
+          const splitContext = extContext.value.split('#');
+          let contextItem = splitContext[0];
+          let contextPath = splitContext.slice(1).join('#');
```

Comment:
This assumes that if there are multiple `#` in the value, the _first_ one is the separator and the remaining ones are in the path.  But... is `#` a valid symbol in a path? I don't _think_ it is (although I'm not 100% sure). But we know `#` is a valid symbol in a URI (which allows fragments). So I'd suggest that we actually should operate on the assumption that the _last_ `#` is the separator.

Reply:
That said, there is a bit of a wrinkle here if a URL has a `#` in the URL itself and there is no path. We'd (incorrectly) assume the last bit is a path. But... maybe you could get around this by putting a `#` at the end of the URL (e.g., _blank_ path)?

Reply:
That is an interesting case. I think your workaround would work fine, in that case. Also, if the author uses an alias, that should also work.